src/lib/services/playground-service.ts

```diff
@@ -0,0 +1,123 @@
+import FeatureToggleService from './feature-toggle-service';
+import { SdkContextSchema } from 'lib/openapi/spec/sdk-context-schema';
+import { InMemStorageProvider, Unleash as UnleashClient } from 'unleash-client';
+import { IUnleashServices } from 'lib/types/services';
+import { ALL } from '../../lib/types/models/api-token';
+import { PlaygroundFeatureSchema } from 'lib/openapi/spec/playground-feature-schema';
+import { Logger } from '../logger';
+import { IUnleashConfig } from 'lib/types';
+import { FeatureConfigurationClient } from 'lib/types/stores/feature-strategies-store';
+import { Operator } from 'unleash-client/lib/strategy/strategy';
+import { once } from 'events';
+
+enum PayloadType {
+    STRING = 'string',
+}
+
+type NonEmptyList<T> = [T, ...T[]];
+
+export const offlineUnleashClient = async (
+    toggles: NonEmptyList<FeatureConfigurationClient>,
+    context: SdkContextSchema,
+    logError: (message: any, ...args: any[]) => void,
+): Promise<UnleashClient> => {
+    const client = new UnleashClient({
+        ...context,
+        appName: context.appName || 'playground', // this is required and might mess people up
```

Comment:
The SDK requires an appName to start up correctly. That means that we must provide an app name if the user does not. Because that can affect the strategies and what toggles are considered enabled, we should think through what exactly we provide here before we merge it. It cannot be an empty string.

A suggestion is to use something that the user is verrrry unlikely to use themselves (e.g. a random string + a uuid or something), but it's still not perfect.

If possible, I'd like to have a way around providing this, but that would require updating the sdk, so it's def not for the first iteration at least. 

What do you think?

Reply:
@FredrikOseberg Got any thoughts on this point?

Reply:
You can use constraints to only enable toggles for certain appNames, right? That means that if you have a constraint that says the name must be `playground` (in the current way it's set up), you might expect the toggle to be false if you don't set the app name. However, if _we_ set it to be `playground` it will be true in this case.

Reply:
The appName needs to be set in the frontend I think. If you are constraining on appName you will never be able to get the right result from the playground if we hardcode it.

Reply:
Does the frontend _require_ you to set the app name? Or rather: should the API require appName to be set on incoming requests?